replica/table.cc

```diff
@@ -634,11 +635,25 @@ table::update_cache(compaction_group& cg, lw_shared_ptr<memtable> m, std::vector
         m->mark_flushed(std::move(new_ssts_ms));
         try_trigger_compaction(cg);
     });
+    std::exception_ptr ex;
     if (cache_enabled()) {
+      try {
         co_return co_await _cache.update(std::move(adder), *m);
-    } else {
-        co_return co_await _cache.invalidate(std::move(adder)).then([m] { return m->clear_gently(); });
+      } catch (...) {
+        ex = std::current_exception();
+      }
+    }
+    co_await _cache.invalidate(std::move(adder));
+    co_await m->clear_gently();
+    if (ex) {
+        // Just report the exception as invalidating the cache is fine as fallback.
+        tlogger.warn("Failed to update the row cache: {}", std::move(ex));
```

Comment:
Why do you go out of your way to delay this log until after `invalidate`?

Unless there is a good reason, it's better to log it immediately, otherwise a different failure between the catch and the print might cause the original exception to be silently lost.

Reply:
if ex is bad_alloc, I'd like to first free memory before printing to prevent the logger itself from running out of memory.

Reply:
For sufficiently short log messages (<8192 bytes) the logger should be using a static buffer and not require allocation. And if the message is longer and the logger does run out of memory, the message should just be truncated.

Reply:
Working around potential bad_alloc in the logger is insane, I think.

Reply:
fair enough for short messages

Reply:
moved logging into catch clause in v3